rego/rego_wasmtarget_test.go

```diff
@@ -89,3 +97,122 @@ func TestPrepareAndEvalWithWasmTargetModulesOnCompiler(t *testing.T) {
 		EvalInput(map[string]int{"x": 1}),
 	}, "[[true]]")
 }
+
+func TestEvalWithContextTimeout(t *testing.T) {
+	ts := httptest.NewServer(http.HandlerFunc(func(_ http.ResponseWriter, r *http.Request) {
+		select {
+		case <-r.Context().Done():
+			// Without this, our test execution would hang waiting for this server to have
+			// served all requests to the end -- unrelated to the behaviour in the client,
+			// so the test would still pass.
+			return
+		case <-time.After(5 * time.Second):
+			return
+		}
+	}))
+	defer ts.Close()
+
+	// This is host function, i.e. it's not implemented natively in wasm,
+	// but calls the topdown function from the wasm instance's execution.
+	// Also, it uses the topdown.Cancel mechanism for cancellation.
+	cidrExpand := `package p
+allow {
+	net.cidr_expand("1.0.0.0/1")
+}`
+
+	// Also a host functiopn, but uses context.Context for cancellation.
+	httpSend := fmt.Sprintf(`package p
+allow {
+	http.send({"method": "get", "url": "%s", "raise_error": true})
+}`,
+		ts.URL)
+	httpSendTopdownError := fmt.Sprintf(`http.send: Get "%s": context deadline exceeded`, ts.URL)
+
+	// This is a natively-implemented function that just takes long.
+	// NOTE(sr): On my machine, it starts exceeding 200ms when passed 1e6,
```

Comment:
This comment is from a time when I hadn't adjusted the timeout to work with CI. I'll drop it before merging.